### PR TITLE
kicad: 7.0.10 -> 8.0.1

### PR DESCRIPTION
**Pull request by evils**
## Description of changes

[7.0.11 release blog post](https://www.kicad.org/blog/2024/02/KiCad-7.0.11-Release/)
[8.0.0 release blog post](https://www.kicad.org/blog/2024/02/Version-8.0.0-Released/)
[8.0.1 release blog post](https://www.kicad.org/blog/2024/03/KiCad-8.0.1-Release)

## Things done

bump 7.0.10 to 7.0.11 (included separately to stay reachable and be backported to 23.11)
bump kicad-unstable to 2024-02-07 (fixes its build, adds dependencies that are now part of 8.0.0, update wrapper env vars)
bump 7.0.11 to 8.0.0 (uses the kicad-unstable new dependencies and env var wrappers)
(also bumps kicad-unstable and kicad-testing, should be almost exactly the same at the moment)

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
  - 4 packages failed to build: kicadAddons.kikit kicadAddons.kikit-library kikit kikit.dist
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by evils** [comment]
ping KiKit maintainers @jfly @matusf
it appears this breaks `kikit` etc.
(kicadAddons.kikit kicadAddons.kikit-library kikit kikit.dist)

**Comment by bgamari** [comment]
Thanks for handling this, @evils!

File: pkgs/applications/science/electronics/kicad/base.nix
Comment (evils) [reply pkgs/applications/science/electronics/kicad/base.nix:100]: there is no associated ticket with the qa_spice failure
the qa_eeschema thing has been resolved
i've got a draft issue on the qa_spice thing
though out-of-band i got the impression they're not very concerned about it
(i kinda don't want to bother them with something that only "fails" in nixpkgs
(it's not actually a negative result, just bad test setup, i think))

i do have the impression there was a way to exclude specific sub-tests in a `qa_*` suite...

if i end up filing an issue i'll add the reference/fix, but i wouldn't want to block merging waiting for this

**Comment by evils** [comment]
i suggest this approach for KiKit:
keep 23.11 on 7.x where stable KiKit will keep working
update KiKit on nixos-unstable to master, where it seems the compatibility is already fixed

**Comment by evils** [comment]
also @jfly i've edited your runtime data path patch's env var to use `KICAD8` to match up with the other env vars, i hope that's ok

**Comment by evils** [comment]
oh and on sway (wayland) with mixed DPI monitors this may have trouble getting an appropriate area
this may be a wxwidgets issue and may be resolved on their master branch, i haven't looked into it for a while...

File: pkgs/applications/science/electronics/kicad/base.nix
Comment (evils) [reply pkgs/applications/science/electronics/kicad/base.nix:100]: [issue reported upstream](https://gitlab.com/kicad/code/kicad/-/issues/17133)
replaced comment with link to that

**Review by jfly — COMMENTED** [review]
Thanks, @evils! For the record, kikit does not seem to be buildable on this branch. How I tested:

```
$ nix-build -E 'let pkgs = import (builtins.fetchTarball { url = "https://github.com/NixOS/nixpkgs/archive/e70e1370a88f86ea989902e140172e9cff0d379a.tar.gz"; }) {}; in pkgs.kicad.override {addons=[pkgs.kicadAddons.kikit pkgs.kicadAddons.kikit-library];}'
...
not ok 38 Dumping preset with plugin
# (in test file /build/source/test/system/panelize.bats, line 301)
#   `kikit panelize --dump preset-plugin.json \' failed
# 01:58:47: Error: Directory '/homeless-shelter' couldn't be created (error 13: Permission denied)
# 01:58:47: Error: Directory '/homeless-shelter/.config/kicad/8.0' couldn't be created (error 2: No such file or directory)
# An error occurred: module 'pcbnewTransition.pcbnew' has no attribute 'EDA_RECT'
# No output files produced
# Traceback (most recent call last):
#   File "/nix/store/79gz22x7k5q7755hzcnz501g0is7fsd3-kikit-1.4.0/lib/python3.11/site-packages/kikit/panelize_ui.py", line 217, in panelize
#     doPanelization(input, output, preset, plugin)
#   File "/nix/store/79gz22x7k5q7755hzcnz501g0is7fsd3-kikit-1.4.0/lib/python3.11/site-packages/kikit/panelize_ui.py", line 266, in doPanelization
#     ki.buildLayout(preset, panel, input, sourceArea)
#   File "/nix/store/79gz22x7k5q7755hzcnz501g0is7fsd3-kikit-1.4.0/lib/python3.11/site-packages/kikit/panelize_ui_impl.py", line 268, in buildLayout
#     substrates = lPlugin.buildLayout(panel, sourceBoard, sourceArea)
#                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
#   File "/build/source/test/resources/testplugin.py", line 33, in buildLayout
#     panel.appendBoard(inputFile, pcbnew.wxPointMM(0, 0), sourceArea)
#   File "/nix/store/79gz22x7k5q7755hzcnz501g0is7fsd3-kikit-1.4.0/lib/python3.11/site-packages/kikit/panelize.py", line 991, in appendBoard
#     drawings = collectItems(board.GetDrawings(), enlargedSourceArea)
#                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
#   File "/nix/store/79gz22x7k5q7755hzcnz501g0is7fsd3-kikit-1.4.0/lib/python3.11/site-packages/kikit/common.py", line 83, in collectItems
#     return list([x for x in boardCollection if fitsIn(x.GetBoundingBox(), sourceArea)])
#                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
#   File "/nix/store/79gz22x7k5q7755hzcnz501g0is7fsd3-kikit-1.4.0/lib/python3.11/site-packages/kikit/common.py", line 83, in <listcomp>
#     return list([x for x in boardCollection if fitsIn(x.GetBoundingBox(), sourceArea)])
#                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
#   File "/nix/store/79gz22x7k5q7755hzcnz501g0is7fsd3-kikit-1.4.0/lib/python3.11/site-packages/kikit/common.py", line 49, in fitsIn
#     assert isinstance(what, (BOX2I, VECTOR2I, pcbnew.wxPoint, pcbnew.EDA_RECT))
#                                                               ^^^^^^^^^^^^^^^
# AttributeError: module 'pcbnewTransition.pcbnew' has no attribute 'EDA_RECT'
ok 39 Present a single board # skip PcbDraw is not available, skipping
ok 40 Steel stencils
ok 41 Steel stencils with cutout
ok 42 Printed stencils
make: *** [Makefile:40: test-system] Error 1
```

I see this is fixed upstream in https://github.com/yaqwsx/KiKit/commit/c920ba3ca87d7878a330e34297ace922b26c0d84, which was released just 16 hours ago in kikit 1.5.0. Looking at the ryantm update logs, I think it just hasn't realized there's a new version of kikit yet: https://r.ryantm.com/log/kikit/ (the last run was 3 days ago on 2024-02-23). I did try upgrading locally, and I see a new test failure, though. If it's ok, I'd like to wait a day to see if the update bot sends in an update PR (I'm a quite new maintainer of nixpkgs and would like to learn how to fix a broken auto-update PR).

**Comment by evils** [comment]
i don't believe the bot submits PRs if the package fails to build :)

**Comment by jfly** [comment]
Ah, OK. How am I as a maintainer supposed to know that happened so I can intervene?

**Comment by evils** [comment]
good question, i don't believe it emits such indications
i suppose that's in part to avoid spamming people

one thing you can do is run this script once in a while
`nix-shell maintainers/scripts/update.nix --argstr maintainer jfly`
i don't think that's nearly as extensive as what the bot does
it does seem to get everything you maintain minus kikit/solidpython

**Comment by jfly** [comment]
Gotcha. That's a bummer, I thought I just had to look at my email to keep things up to date.

For the record, I've created https://github.com/NixOS/nixpkgs/pull/291734 to get kikit upgraded. If we can get that merged first, then there shouldn't be a window of time in `master` where kikit is broken.

**Review by jfly — APPROVED** [review]
Code LGTM, but I still can't successfully test out kikit on this branch because it doesn't have the latest changes from master.

I tried checking out this PR locally and rebasing on top of the latest in `master` and got a conflict in pkgs/development/libraries/log4cplus/default.nix that I'm stumped by.

**Comment by kirillrdy** [comment]
@jfly if you run `nixpkgs-review pr 291134`

it will build you a merge of master and this PR

**Comment by jfly** [comment]
Thanks, @kirillrdy. I just manually tested out kikit with this PR, and things seem to work!

**Comment by kirillrdy** [comment]
Result of `nixpkgs-review pr 291134` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>python312Packages.kicad</li>
    <li>python312Packages.pcbnew-transition</li>
    <li>python312Packages.pcbnew-transition.dist</li>
  </ul>
</details>
<details>
  <summary>13 packages built:</summary>
  <ul>
    <li>kicad</li>
    <li>kicad-small</li>
    <li>kicad-testing</li>
    <li>kicad-testing-small</li>
    <li>kicad-unstable</li>
    <li>kicad-unstable-small</li>
    <li>kicadAddons.kikit</li>
    <li>kicadAddons.kikit-library</li>
    <li>kikit</li>
    <li>kikit.dist</li>
    <li>python311Packages.kicad</li>
    <li>python311Packages.pcbnew-transition</li>
    <li>python311Packages.pcbnew-transition.dist</li>
  </ul>
</details>


3.12 build fails because of diffimg which also fails on master

**Comment by evils** [comment]
oh, i maintain `diffimg` -_-

**Comment by kiranshila** [comment]
Perhaps this PR can also cover the recent bugfix release as well? https://www.kicad.org/blog/2024/03/KiCad-8.0.1-Release/

**Comment by evils** [comment]
pushed 8.0.1

**Comment by kirillrdy** [comment]
Result of `nixpkgs-review pr 291134` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages failed to build:</summary>
  <ul>
    <li>kicad</li>
    <li>kicad-small</li>
    <li>kicad-testing</li>
    <li>kicad-testing-small</li>
    <li>kicad-unstable</li>
    <li>kicad-unstable-small</li>
    <li>kicadAddons.kikit</li>
    <li>kicadAddons.kikit-library</li>
    <li>kikit</li>
    <li>kikit.dist</li>
    <li>python311Packages.kicad</li>
    <li>python311Packages.pcbnew-transition</li>
    <li>python311Packages.pcbnew-transition.dist</li>
    <li>python312Packages.kicad</li>
    <li>python312Packages.pcbnew-transition</li>
    <li>python312Packages.pcbnew-transition.dist</li>
  </ul>
</details>

**Comment by evils** [comment]
it doesn't appear as if aarch64-linux is supported by upstream
not sure how best to reflect this in the package here
i'm inclined to skip the `qa_cli` test for `aarch64-linux` and hope that if there is an issue with the actual app a Nix user reports the issue to nixpkgs
(i have an aarch64 VPS available, but nothing i can practically check the GUI on)

**Comment by lukaslihotzki** [comment]
I can confirm that the Kicad (both 8.0.1 and testing 2024-03-16) GUI works fine on aarch64 NixOS. I disabled the tests completely (`doInstallCheck = false;`). Although this should be independent of `qa_cli` which checks the `kicad-cli` CLI tool.

Additionally, I get this build error when doing `nixpkgs-review pr 291134`: `ModuleNotFoundError: No module named 'distutils'` when building `python3.12-wxpython-4.2.1`.

**Comment by lukaslihotzki** [comment]
The `qa_cli` test fails because of some image differences and because the items in the orcadpcb2 output are unexpectedly reordered. So, the item ordering of orcadpcb2 (which is determined by an rtree) seems to be architecture-specific. I don't know if this is a bug in rtree or if the test is too strict, but this test fail should be harmless otherwise. I haven't looked at the image differences yet.

**Comment by evils** [comment]
i've added disabling the qa_cli test on aarch64-linux

**Comment by kirillrdy** [comment]
@evils thank you for PR
feel free to tag me for any fallout

**Comment by evils** [comment]
note for completeness: someone from upstream successfully ran the qa_cli test on ubuntu on aarch64
but confirmed it's not an officially supported platform

**Pull request merged**
